src/babashka/fs.cljc

```diff
@@ -1125,3 +1128,51 @@
   (if win?
     (-> f as-path .toUri .getPath)
     (str f)))
+
+(defn- xdg-path-from-env-var
+  "Yields value of environment variable `k` as path if it's an absolute
+  path. Else `nil`."
+  [k]
+  (some-> (get-env k)
+          (#(when (absolute? %) %))
+          (path)))
+
+(def ^:private xdg-type->env-var&default-path
+  (delay {:config ["XDG_CONFIG_HOME" (path (home) ".config")]
+          :cache  ["XDG_CACHE_HOME" (path (home) ".cache")]
+          :data   ["XDG_DATA_HOME" (path (home) ".local" "share")]
+          :state  ["XDG_STATE_HOME" (path (home) ".local" "state")]}))
+
+(defn- xdg-home-for [k]
+  (let [[env-var default-path] (@xdg-type->env-var&default-path k)]
+    (or
+     (xdg-path-from-env-var env-var)
+     default-path)))
+
+(defn xdg-config-path
```

Comment:
Why not just add `(xdf-config-home)` and leave the construction of paths to `path`?

``` clojure
(fs/file (fs/xdg-config-home) "foo" "bar")
```

Choosing the `& child` arguments doesn't allow us to add an option parameter if we need to, for example.

Reply:
I started with that idea and later figured that you mostly combine this with the name of the app (and a specific file).
I see the problem with any future options.   
An alternative could be to allow for zero or one argument `(fs/xdg-config-home "clj-kondo")`. WDYT?

Reply:
I'd say let's start out the simplest way possible and leave the path construction to `path` and `file`, although it's tempting to make it like this now.

Reply:
I'm looking forward replacing this code:

https://github.com/babashka/pods/blob/75c2216649bf5caf3ae41e7b8fc202e45b090ea9/src/babashka/pods/impl/resolver.clj#L121-L122

Reply:
> I'm looking forward replacing this code:

I've seen a lot of code like this: use xdg-env-var if present, else use home-dot-app (ie non-xdg legacy).

To allow users to be xdg-compliant (ie when they move `~/.basbashka/pods` to `~/.local/share/basbashka/pods`), the following works I think:
```clojure
(fs/path
  (or (some-> (System/getenv "BABASHKA_PODS_DIR") (fs/path))
      (some-> (System/getenv "XDG_DATA_HOME") (fs/path ".babahaska" "pods") (#(when (fs/exists? %) %)))
      (some-> (fs/path (fs/home) ".babashka" "pods") (#(when (fs/exists? %) %)))
      (fs/path (fs/xdg-data-home) "babashka" "pods"))
  "repository")
```

Reply:
Should `(System/getenv "XDG_DATA_HOME")` be `(System/getenv "HOME")` in your example?

Reply:
Makes sense. And for new project, the middle clause can be left out :)